tests/service/CSharpProjectAnalysis.fs

```diff
@@ -0,0 +1,90 @@
+    let options =
+        checker.GetProjectOptionsFromCommandLineArgs(projFileName,
+            [| yield "--debug:full" 
+               yield "--define:DEBUG" 
+               yield "--optimize-" 
+               yield "--out:" + dllName
+               yield "--doc:test.xml" 
+               yield "--warn:3" 
+               yield "--fullpaths" 
+               yield "--flaterrors" 
+               yield "--target:library" 
+               for dllFile in dllFiles do
+                 yield "-r:"+dllFile
+               for libDir in libDirs do
+                 yield "-I:"+libDir
+               yield! otherFlags
+               yield fileName1 |])
+    let results = checker.ParseAndCheckProject(options) |> Async.RunSynchronously
+    if results.HasCriticalErrors then
+        let builder = new System.Text.StringBuilder()
+        for err in results.Errors do
+            builder.AppendLine(sprintf "**** %s: %s" (if err.Severity = Microsoft.FSharp.Compiler.Severity.Error then "error" else "warning") err.Message)
+            |> ignore
+        failwith (builder.ToString())
+
+    results.ProjectContext.GetReferencedAssemblies()
+        |> List.map(fun x -> x.SimpleName, x)
+        |> dict
+
+[<Test>]
+let ``Test that csharp references are recognized as such`` () = 
+    let csharpAssembly = typeof<CSharpClass>.Assembly.Location
+    let table = getProjectReferences([csharpAssembly], None, None)
+    let ass = table.["CSharp_Analysis"]
+    match ass.Contents.Entities |> Seq.tryFind (fun e -> e.DisplayName = "CSharpClass") with
+    | Some found ->
+        // this is no F# thing
+        found.IsFSharp |> shouldEqual false
+        
+        // Check that we have members
+        let members = found.MembersFunctionsAndValues |> Seq.map (fun e -> e.CompiledName, e) |> dict
+        members.ContainsKey ".ctor" |> shouldEqual true
+        members.ContainsKey "Method" |> shouldEqual true
+        members.ContainsKey "Property" |> shouldEqual true
+        members.ContainsKey "Event" |> shouldEqual true
+        members.ContainsKey "InterfaceMethod" |> shouldEqual true
+        members.ContainsKey "InterfaceProperty" |> shouldEqual true
+        members.ContainsKey "InterfaceEvent" |> shouldEqual true
+
+        //// Check that we get xml docs
+        //String.IsNullOrWhiteSpace(members.[".ctor"].XmlDocSig) |> shouldEqual false
+        //String.IsNullOrWhiteSpace(members.["Method"].XmlDocSig) |> shouldEqual false
+        //String.IsNullOrWhiteSpace(members.["Property"].XmlDocSig) |> shouldEqual false
+        //String.IsNullOrWhiteSpace(members.["Event"].XmlDocSig) |> shouldEqual false
+        //String.IsNullOrWhiteSpace(members.["InterfaceMethod"].XmlDocSig) |> shouldEqual false
+        //String.IsNullOrWhiteSpace(members.["InterfaceProperty"].XmlDocSig) |> shouldEqual false
+        //String.IsNullOrWhiteSpace(members.["InterfaceEvent"].XmlDocSig) |> shouldEqual false
+
+        ()
+    | None -> ()
```

Comment:
Correct :). While I noticed this shortly after pushing it, it's good to have a reminder comment, thanks. I guess it is no big deal for now (as that part is never executed), but I will fix that after getting an overall review/comment on the code.